install/kubernetes/helm/subcharts/gateways/templates/clusterrole.yaml

```diff
@@ -15,6 +15,25 @@ rules:
   resources: ["virtualservices", "destinationrules", "gateways"]
   verbs: ["get", "watch", "list", "update"]
 ---
+{{- if eq $key "istio-ingressgateway" }}
+{{- if .Values.ingressGatewayAgent.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
```

Comment:
Strong -1 on using ClusterRole for that. I mentioned this in the design doc as well. 
Ingress should only see secrets in same namespace.

Reply:
Incidentally this will also break @linsun  a-la-carte ingress gateway support, and my 1.2 a-la-carte.

Reply:
Added install/kubernetes/helm/subcharts/gateways/templates/role.yaml and install/kubernetes/helm/subcharts/gateways/templates/rolebindings.yaml to let ingress-sds-agent watches secrets in the same namespace.